src/compass/utils/helpers.py

```diff
@@ -348,10 +348,17 @@ def open_raster(filename, band=1):
         raise FileNotFoundError(err_str)
 
     try:
+        # The pythonic execption handling for GDAL can be turned on / off
+        # The flag of which can be identified by `gdal.GetUseExceptions()`
+        # In pythonic exception halding, `gdal.Open()` will raise `RuntimeError`
+        # In traditional GDAL, the function does not raise exception buy will return `None`,
+        # and the attempts to call GDAL methods will raise `AttributeError`
+
         ds = gdal.Open(filename, gdal.GA_ReadOnly)
         arr = ds.GetRasterBand(band).ReadAsArray()
         return arr
-    except AttributeError:
+
+    except Exception:
```

Comment:
```suggestion
    except Exception (RuntimeError, AttributeError):
```
Did the above not work? I was able to get it work with the following toy problem:
```
In [30]: def raise_one_or_other():
    ...:     rando = np.random.rand()
    ...:     if rando > 0.5:
    ...:         raise RuntimeError(f"RuntimeError: {rando}> 0.5")
    ...:     else:
    ...:         raise AttributeError(f"AttributeError: {rando} <= 0.5")

In [31]: try:
    ...:     raise_one_or_other()
    ...: except (RuntimeError, AttributeError) as err:
    ...:     print(type(err), err.args, isinstance(err, AttributeError))
    ...: 
<class 'AttributeError'> ('AttributeError: 0.1704103346669137 <= 0.5',) True

In [32]: try:
    ...:     raise_one_or_other()
    ...: except (RuntimeError, AttributeError) as err:
    ...:     print(type(err), err.args, isinstance(err, AttributeError))
    ...: 
<class 'RuntimeError'> ('RuntimeError: 0.8876243545798816> 0.5',) False

```

Geoff recommended that [`gdal.UseExceptions` be used](https://github-fn.jpl.nasa.gov/isce-3/isce/issues/922). It's used on occasion in ISCE3 and it's called immediately after `from osgeo import gdal`. If this recommendation is adopted in this file, then only the `RuntimeError` needs to be caught.

Whatever the case I would prefer catching specific errors so unexpected errors can be raised.

I forgot why we don't exclusively `gdal.OpenEx` in the `except` block and avoid `gdal.Open`?